### PR TITLE
Guidance for AggressiveInlining and AggressiveOptimization

### PR DESCRIPTION
**Pull request by danmoseley**
per comments in https://github.com/dotnet/runtime/pull/90504

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by EgorBo — APPROVED** [review]

**Comment by danmoseley** [comment]
@docs this should be ready

File: xml/System.Runtime.CompilerServices/MethodImplOptions.xml [hunk 1]
```diff
@@ -144,9 +147,8 @@
...
-          <para>Methods with this attribute may not have optimal code generation. They bypass the first tier of <a href="/dotnet/core/runtime-config/compilation#tiered-compilation">Tiered Compilation</a> and therefore can't benefit from optimizations that rely on tiering, for example, <a href="/dotnet/core/runtime-config/compilation#profile-guided-optimization">Dynamic PGO</a> or optimizations based on initialized classes.</para>
+          <para>The method contains code that should always be optimized for performance.</para>
+          <para>It is rarely appropriate to use this attribute. Methods that apply this attribute will bypass the first tier of <a href="/dotnet/core/runtime-config/compilation#tiered-compilation">Tiered Compilation</a> and therefore will not benefit from optimizations that rely on tiered compilation, for example, <a href="/dotnet/core/runtime-config/compilation#profile-guided-optimization">Dynamic PGO</a> or optimizations based on initialized classes. It may also increase memory use. Always measure performance to ensure it is helpful to apply this attribute.</para>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/MethodImplOptions.xml:151]:
```suggestion
          <para>It's rarely appropriate to use this attribute. Methods that apply this attribute bypass the first tier of <a href="/dotnet/core/runtime-config/compilation#tiered-compilation">tiered compilation</a> and therefore don't benefit from optimizations that rely on tiered compilation. Those optimizations include <a href="/dotnet/core/runtime-config/compilation#profile-guided-optimization">dynamic PGO</a> and optimizations based on initialized classes. Use of this attribute might also increase memory use. Always measure performance to ensure it's helpful to apply this attribute.</para>
```

File: xml/System.Runtime.CompilerServices/MethodImplOptions.xml [hunk 2]
```diff
@@ -113,7 +113,10 @@
       </ReturnValue>
       <MemberValue>256</MemberValue>
       <Docs>
-        <summary>The method should be inlined if possible.</summary>
+        <summary>
+          <para>The method should be inlined if possible.</para>
+          <para>Unnecessary use of this attribute can reduce performance. It may cause implementation limits to be encountered that will cause slower code to be generated. Always measure performance to ensure it is helpful to apply this attribute.</para>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/MethodImplOptions.xml:118]:
```suggestion
          <para>Unnecessary use of this attribute can reduce performance. The attribute might cause implementation limits to be encountered that will result in slower generated code. Always measure performance to ensure it's helpful to apply this attribute.</para>
```

Comment (gewarren) [reply xml/System.Runtime.CompilerServices/MethodImplOptions.xml:118]: Removed double "cause".

**Review by gewarren — APPROVED** [review]

**Pull request merged**